### ISSUE TITLE
Ability to remove tags field from output broke in 5.0.1 and above

### ISSUE THREAD

**Issue by ronnocol** [elastic/logstash#6429]
I'm trying to output to a very picky HTTP API, and the existence of a field in the JSON that is posted causes it confusion. I originally did the integration work with 5.0.0 (latest at the time) but then when we went to deploy, grabbed the latest version (5.1.0) I then checked every iteration back to 5.0.0 and found that the ability to remove the 'tags' field broke in 5.0.1.

```text
[ronnocol@tarpon logstash-5.0.0]$ bin/logstash -e 'filter { mutate { remove_field => ["tags"] }}'
Sending Logstash logs to /home/loconnor/projects/logstash-5.0.0/logs which is now configured via log4j2.properties.
The stdin plugin is now waiting for input:
[2016-12-15T09:52:02,658][INFO ][logstash.pipeline        ] Starting pipeline {"id"=>"main", "pipeline.workers"=>8, "pipeline.batch.size"=>125, "pipeline.batch.delay"=>5, "pipeline.max_inflight"=>1000}
[2016-12-15T09:52:02,675][INFO ][logstash.pipeline        ] Pipeline main started
[2016-12-15T09:52:02,727][INFO ][logstash.agent           ] Successfully started Logstash API endpoint {:port=>9600}
lsdkfj
{
    "@timestamp" => 2016-12-15T17:52:07.282Z,
      "@version" => "1",
          "host" => "tarpon.engr.blah.com",
       "message" => "lsdkfj",
          "type" => "stdin"
}
```
Notice that the "tags" field has been properly removed.

In 5.0.1 and above, this same test no longer works:

```text
[loconnor@tarpon logstash-5.0.1]$ bin/logstash -e 'filter { mutate { remove_field => ["tags"] }}' 
Sending Logstash's logs to /home/loconnor/projects/logstash-5.0.1/logs which is now configured via log4j2.properties
The stdin plugin is now waiting for input:
[2016-12-15T10:30:44,329][INFO ][logstash.pipeline        ] Starting pipeline {"id"=>"main", "pipeline.workers"=>8, "pipeline.batch.size"=>125, "pipeline.batch.delay"=>5, "pipeline.max_inflight"=>1000}
[2016-12-15T10:30:44,344][INFO ][logstash.pipeline        ] Pipeline main started
[2016-12-15T10:30:44,390][INFO ][logstash.agent           ] Successfully started Logstash API endpoint {:port=>9600}
asdf
{
    "@timestamp" => 2016-12-15T18:30:50.053Z,
      "@version" => "1",
          "host" => "tarpon.engr.blah.com",
       "message" => "asdf",
          "type" => "stdin",
          "tags" => []
}
[2016-12-15T10:30:54,365][WARN ][logstash.agent           ] stopping pipeline {:id=>"main"}
```

The existence of extra fields in the JSON causes some picky HTTP APIs to fail.

**Comment by u0x01** [comment]
I have this problem too

**Comment by djschny** [comment]
Duplicate of https://github.com/elastic/logstash/issues/6384 ?

I have this problem as well and it causes my documents to fail loading when using strict mappings.

**Comment by jpyth** [comment]
Hi, is the issue really failure to remove or the fact that it's automatically created to start with(which you then apparently can't remove)?

I found the same issue when upgrading from 2.x to 5.1. Tags was 'automatically' created but with an empty array([]); this caused my indexing to misbehave as I do a if [tags] to direct tagged documents to a unique index vs non.

I downgraded to 5.0 after reading this report and the mysterious 'empty' tags field disappeared. I didn't try 5.0.1 or 5.0.2 to confirm when it appeared; but i suspect i'd see the same as you.

Should I open a new issue for this; or is this considered the same problem?

**Comment by colinsurprenant** [comment]
Yes, duplicate of #6384 
Looking into this.

**Comment by colinsurprenant** [comment]
most probably a regression caused by #6177 which indicates we also miss test coverage for this edge case.

**Comment by colinsurprenant** [comment]
pushed fix #6477

**Comment by colinsurprenant** [comment]
fixed by #6477, will be included in 5.1.2

**Comment by colinsurprenant** [comment]
and thanks @ronnocol for reporting and @u0x01 @djschny @jpyth for following up!
